src/data/shared_schemas.ts

```diff
@@ -0,0 +1,16 @@
+import { JSONSchemaType } from 'ajv';
+import { Amount, AmountType, AmountUnit } from './types/amount';
+
+export const AMOUNT_SCHEMA: JSONSchemaType<Amount> = {
```

Comment:
I think this can go in `amount.ts`; there are both schemas and type definitions in there.